tests/mdev_configuration_allocation_test.go

```diff
@@ -56,18 +57,45 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Eventually(ThisPod(testPod), 3*time.Minute, 5*time.Second).Should(BeInPhase(k8sv1.PodSucceeded))
 	}
-	checkAllMDEVRemoved := func() {
-		check := fmt.Sprintf(`set -x
+
+	var latestPod *k8sv1.Pod
+	waitForPod := func(fetchPod func() (*k8sv1.Pod, error)) wait.ConditionFunc {
+		return func() (bool, error) {
+
+			latestPod, err = fetchPod()
+			if err != nil {
+				return false, err
+			}
+
+			phase := latestPod.Status.Phase
+
+			if phase == k8sv1.PodFailed {
+				return true, nil
+			}
+			return phase == k8sv1.PodSucceeded, nil
+
+		}
+	}
+
+	checkAllMDEVRemoved := func() (*k8sv1.Pod, error) {
+		check := `set -x
 		files_num=$(ls -A /sys/bus/mdev/devices/| wc -l)
 		if [[ $files_num != 0 ]] ; then
 		  echo "failed, not all mdevs removed"
 		  exit 1
 		fi
-	        exit 0`)
-		testPod := tests.RenderPod("test-pod", []string{"/bin/bash", "-c"}, []string{check})
+	        exit 0`
+		testPod := tests.RenderPod("test-all-mdev-removed", []string{"/bin/bash", "-c"}, []string{check})
 		testPod, err = virtClient.CoreV1().Pods(util.NamespaceTestDefault).Create(context.Background(), testPod, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
-		Eventually(ThisPod(testPod), 120).Should(BeInPhase(k8sv1.PodSucceeded))
+
+		err := wait.PollImmediate(time.Second, 2*time.Minute, waitForPod(ThisPod(testPod)))
+		return latestPod, err
```

Comment:
This `err` seems to be never checked. Can we add it, please? Also it seems not to be necessary to return it, since IIUC it it not used from the caller.

Reply:
This will be checked in `Eventually`